Add models for SequencingReads and *PathogenGenomes

ttung opened the pull request:
### Description
Enums for sequencing instrument and sequencing protocol.  Entities for the sequencing reads and pathogen genomes.

Added a test.

Depends on #74, #75

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
ran test!

phoenixAja commented on src/py/aspen/database/models/sequences.py:
```diff
@@ -0,0 +1,145 @@
+    RESTRICTED_ACCESS = "Restricted Access"
+
+
+# Create the enumeration table
+# Pass your enum class and the SQLAlchemy declarative base to enumtables.EnumTable
```
i think it's safe to remove these repetitive comments

phoenixAja replied:
nvrmind, see below

phoenixAja commented on src/py/aspen/database/models/sequences.py:
```diff
@@ -0,0 +1,145 @@
+    ARTIC_V3 = "artic_v3"
+
+
+# Create the enumeration table
+# Pass your enum class and the SQLAlchemy declarative base to enumtables.EnumTable
```
remove

phoenixAja replied:
oh woops for a second i thought i was looking at the migration file, i changed my mind keep these comments in!

phoenixAja commented on src/py/aspen/database/models/sequences.py:
```diff
@@ -0,0 +1,145 @@
+        nullable=False,
+    )
+
+    s3_bucket = Column(String, nullable=False)
+    s3_key = Column(String, nullable=False)
```
just a QQ why not store the bucket and the key together? Since they'll always be used in conjunction with each other anyways?

ttung replied:
I like my data decomposed.  Would rather compose URIs than to parse them.

phoenixAja commented on src/py/aspen/database/models/sequences.py:
```diff
@@ -0,0 +1,145 @@
+
+    # statistics for the pathogen genome
+    num_unambiguous_sites = Column(Integer, nullable=False)
+    num_n = Column(Integer, nullable=False)
+    num_mixed = Column(Integer, nullable=False)
```
it would be nice to find more descriptive names for these fields, I know the bioinformatics people like these names but for me i always have to mentally try to recall what these statistics are referring to. Not saying i want you to change this, just noting.

ttung replied:
@jackkamm Can you help with proper documentation for these fields?

phoenixAja commented on src/py/aspen/database/models/tests/test_sequences.py:
```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+
+from ..physical_sample import PhysicalSample
+from ..sequences import (
+    SequencingInstrumentType,
+    SequencingProtocolType,
+    SequencingReads,
+    UploadedPathogenGenome,
+)
+from ..usergroup import Group
```
i personally don't like using the `..` i think we should replace with the package name when we start using that syntax

ttung replied:
I generally agree with you, but I like it for intra-package imports.  I consider tests for a package an intra-package import.

phoenixAja replied:
why do you like it for intra-package imports? Does it seem less redundant?

ttung replied:
If I refactor, it's less stuff that changes.

phoenixAja reviewed (changes requested):
only comment is replacing .. with package name in test files

Commit:
Add models for SequencingReads and *PathogenGenomes

### Description
Enums for sequencing instrument and sequencing protocol.  Entities for the sequencing reads and pathogen genomes.

Added a test.

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
ran test!
